packages/react-native-editor/__device-tests__/pages/editor-page.js

```diff
@@ -570,6 +587,57 @@ class EditorPage {
 		return await typeString( this.driver, textViewElement, text, clear );
 	}
 
+	// =============================
+	// Search Block functions
+	// =============================
+
+	async getSearchBlockChild( testKeys ) {
+		let locator = `//XCUIElementTypeOther[@name="${ testKeys.ios }"]`;
+		if ( isAndroid() ) {
+			locator = `//android.view.View[starts-with(@content-desc, "${ testKeys.android }")]/android.widget.EditText`;
+		}
+
+		return await this.driver.elementByXPath( locator );
```

Comment:
All very good points. The problem with `elementByAccessibilityId` is that it's an *exact match* locator strategy, but the `accessibilityLabel`s used in the Search block are highly customized for an optimal screenreader experience. This means the label is dynamic and changes depending on the value of the field. Using this method would require the tests to always know the value of each element in the block which could lead to flaky or easily broken tests later down the road. 

To address this I decided to change it up a bit. I added some extra logic to the main search block code to set `accessibilityLabel` to a testID when a screenreader is not actively enabled (ddc4f3dfaf499abbfc1d68cc44a111c1a7c4f4d0 and ddc4f3dfaf499abbfc1d68cc44a111c1a7c4f4d0). I also had to clear out `accessibilityHint` since that gets appended to the `accessibilityLabel` and then used as the `content-desc` in Android. Unfortunately `accessibilityLabel` can only be set on the parent container (`<View/>`) so in order to update the text of an element during Android tests I used `elementByClassName()` which helped to grab the child element needed. Tests updated in e4f0aea48727406195861ea4cf4c782836589f1e. 

The settings portion of the tests I've decided to leave as-is since my code doesn't control the `accessibilityLabel` and it too is dynamic and includes the value of the setting at the time of access. I didn't want to introduce complexity that would later lead to issues. I did some research and it looks like once we upgrade `react-native` to > 0.65, `testID` will work on both Android and iOS which will simplify access to elements without worrying about effecting accessibility. It would probably make more sense to then go back and update the code.